Null Pointer exception occurs when installing application through the server

**Description:**
Null Pointer exception occurs when installing application.
```
curl -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' --header 'Authorization: Bearer <ACCESS-TOKEN>' -d '{
  "deviceIDs": [
    "1567393930303"
  ],
  "operation": {
    "appIdentifier": "<APP-IDENTIFIER>",
    "type": "enterprise",
    "url": "<HOST>/helloworld.apk"
  }
}' 'https://<HOST>:<PORT>/api/device-mgt/android/v1.0/admin/devices/install-application'
```
```
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
                                                          at org.wso2.iot.agent.events.listeners.ApplicationStateListener.applyEnforcement(ApplicationStateListener.java:152)
                                                          at org.wso2.iot.agent.events.listeners.ApplicationStateListener.onReceive(ApplicationStateListener.java:94)
                                                          at android.app.ActivityThread.handleReceiver(ActivityThread.java:2610)
                                                          at android.app.ActivityThread.access$1700(ActivityThread.java:152) 
                                                          at android.app.ActivityThread$H.handleMessage(ActivityThread.java:138
```
**Affected Product Version:**
3.1.0-Update5

**OS, DB, other environment details and versions:**    
Ubuntu 16.04, H2, Single IOT node 

**Steps to reproduce:**
Enroll an android device.
Install an application

--- a/client/client/src/main/java/org/wso2/iot/agent/events/listeners/ApplicationStateListener.java
+++ b/client/client/src/main/java/org/wso2/iot/agent/events/listeners/ApplicationStateListener.java
@@ -25,6 +25,7 @@ import android.content.Context;
 import android.content.Intent;
 import android.content.IntentFilter;
 import android.os.Build;
+import android.text.TextUtils;
 import android.util.Log;
 import org.json.JSONArray;
 import org.json.JSONException;
@@ -149,9 +150,9 @@ public class ApplicationStateListener extends BroadcastReceiver implements Alert
         String ownershipType = Preference.getString(context, Constants.DEVICE_TYPE);
 
         String restrictionList = null;
-        if(!whiteListAppsPref.equals("")) {
+        if(!TextUtils.isEmpty(whiteListAppsPref)) {
             restrictionList = whiteListAppsPref;
-        } else if(!blackListAppsPref.equals("")) {
+        } else if(!TextUtils.isEmpty(blackListAppsPref)) {
             restrictionList = blackListAppsPref;
         }
 

